File: 6. Introduction-to-Recursion/Tower_of_Hanoi.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) {
        // write your code here
        Scanner scn = new Scanner(System.in);
        int n = scn.nextInt();
        int t1id = scn.nextInt();
        int t2id = scn.nextInt();
        int t3id = scn.nextInt();
        toh(n, t1id, t2id, t3id);
    }

    public static void toh(int n, int t1id, int t2id, int t3id) {
        if (n == 0) {
            // System.out.println(n + "[" + t1id + " -> " + t2id + "]");
            return;
        }
        toh(n - 1, t1id, t3id, t2id);
        // toh(1,t1id,t2id,t3id);
        System.out.println(n + "[" + t1id + " -> " + t2id + "]");
        toh(n - 1, t3id, t2id, t1id);
    }

}

File: Level-1/6. Introduction-to-Recursion/cpp/Tower_of_Hanoi.cpp
#include "bits/stdc++.h"
using namespace std;

void toh(int n,int t1id, int t2id, int t3id){
	if (n == 0) {
		return;
	}
	toh(n - 1, t1id, t3id, t2id);
	// toh(1,t1id,t2id,t3id);
	cout << n << "[" << t1id << " -> " << t2id << "]" << endl;
	toh(n - 1, t3id, t2id, t1id);
}

int main(){
	int n,t1,t2,t3;
	cin >> n >> t1 >> t2 >> t3;
	toh(n,t1,t2,t3);
}
